### DISCUSSION TITLE
Using midpoint integration rule

### DISCUSSION THREAD

**Discussion by salaudeen-ya** [idaholab/moose#23775] [Q&A General] [answered]
Hello

My specific problem requires using the midpoint integration rule for the finite element.

Any idea how to do that?

Thank you.

**Comment by GiudGiud** [comment, marked as answer]
You can look at specifying the quadrature
https://mooseframework.inl.gov/source/actions/SetupQuadratureAction.html
This page is missing the type parameter, which can be set to any of those
https://github.com/idaholab/moose/blob/4d27e9a6206a7402e2c9078495f1fdf35faa8a7c/framework/src/actions/SetupQuadratureAction.C#L20

Guillaume

**Reply by salaudeen-ya** [reply]
Thank you.

**Reply by salaudeen-ya** [reply]
Quick notice:

The custom block in the Quadrature is not effectively pointing at the specified block. I was trying to make sure its working properly in my app by testing it on a simple input file specified in the documentation. I put 200 (as below) and the solver still works - no error, the same as when no block is specified (` `). Is this behavior normal?

```
[Mesh]
  [gmg]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 1
    ny = 2
    xmin = 0
    xmax = 1
    ymin = 0
    ymax = 2
  []

  [bottom]
    type = SubdomainBoundingBoxGenerator
    input = gmg
    block_id = 1
    bottom_left = '0 0 0'
    top_right = '1 1 0'
  []
  [top]
    type = SubdomainBoundingBoxGenerator
    input = bottom
    block_id = 2
    bottom_left = '0 1 0'
    top_right = '1 2 0'
  []
  [middle]
    type = SideSetsBetweenSubdomainsGenerator
    input = top
    primary_block = 1
    paired_block = 2
    new_boundary = middle
  []
[]

[Problem]
  type = FEProblem
  solve = false
[]

[Executioner]
  type = Steady
  [Quadrature]
    custom_blocks = '1 200'
    custom_orders = 'first second'
  []
[]

[Outputs]
  execute_on = 'timestep_end'
[]
```

**Reply by GiudGiud** [reply]
we dont have errors on all objects that take subdomain. 
We did it like this because sometimes subdomains appear during the simulation. 

This could be improved but it should not be limiting you right now
